java-test-samples/async-lambda-dynamodb/src/test/java/com/amazon/aws/sample/TestAsyncTransformation.java

```diff
@@ -0,0 +1,216 @@
+                    destinationBucketName = output.outputValue();
+                } else if ("RecordTransformationTable".equals(output.outputKey())) {
+                    recordTransformationTable = output.outputValue();
+                }
+            }
+        }
+        assertNotNull(sourceBucketName, "sourceBucketName may not be null");
+        assertNotNull(destinationBucketName, "destinationBucketName may not be null");
+    }
+
+    @Test
+    public void testAsyncTransformation_when_txt() {
+        String key = String.format("%s.txt", UUID.randomUUID().toString());
+
+        PutObjectRequest putObjectRequest = PutObjectRequest.builder().bucket(sourceBucketName)
+                .key(key).build();
+
+        PutObjectResponse putObjectResponse = s3Client.putObject(putObjectRequest, RequestBody.fromString(MESSAGE));
+        assertNotNull(putObjectResponse, "putObjectResponse may not be null");
+
+        if (recordTransformationTable != null) {
+            try {
+                Callable<String> callable = getContentFromDynamoDB(recordTransformationTable, key);
+
+                String contentFromRecordTransformationTable = callable.call();
+
+                assertEquals(contentFromRecordTransformationTable, contentFromRecordTransformationTable,
+                        String.format("contentFromRecordTransformationTable must be %s instead of %s",
+                                MESSAGE_TO_UPPER_CASE, contentFromRecordTransformationTable));
+            } catch (Exception exception) {
+                fail("An exception occurred and should not have");
+            }
+        }
+    }
+
+    @Test
+    public void testAsyncTransformation_when_not_txt() {
+        String key = String.format("%s.not_txt", UUID.randomUUID().toString());
+
+        PutObjectRequest putObjectRequest = PutObjectRequest.builder().bucket(sourceBucketName)
+                .key(key).build();
+
+        PutObjectResponse putObjectResponse = s3Client.putObject(putObjectRequest, RequestBody.fromString(MESSAGE));
+        assertNotNull(putObjectResponse, "putObjectResponse may not be null");
+
+        if (recordTransformationTable != null) {
+            try {
+                Callable<String> callable = getContentFromDynamoDB(recordTransformationTable, key);
+
+                callable.call();
+
+                fail("It not reach here, an exception should be thrown");
+            } catch (Exception exception) {
+                assertTrue((exception instanceof RecordNotFoundException), "An exception occured and shoud not have");
+            }
+        }
+    }
+
+    private Callable<String> getContentFromDynamoDB(String tableName, String key)
+            throws InternalErrorException, RecordNotFoundException {
+        IntervalFunction intervalFunction = ofExponentialBackoff(3000l, 1.5d);
+
+        RetryConfig retryConfig = RetryConfig.custom().maxAttempts(3).intervalFunction(intervalFunction)
```

Comment:
Can we not override dynamodb client config instead to handle retry instead resiliency4j library? like below -

/**
     * Method to overwrite the default SDK client configuration behavior
     *
     * @return ClientConfiguration with custom timeout values and retry method
     */
    private static ClientConfiguration createDynamoDBClientConfiguration() {
        ClientConfiguration clientConfiguration = new ClientConfiguration()
               .withConnectionTimeout(MyDynamoDBClientParameters.connectionTimeout)
                .withClientExecutionTimeout(MyDynamoDBClientParameters.clientExecutionTimeout)
                .withRequestTimeout(MyDynamoDBClientParameters.requestTimeout)
                .withSocketTimeout(MyDynamoDBClientParameters.socketTimeout)
                .withRetryPolicy(PredefinedRetryPolicies
                        .getDynamoDBDefaultRetryPolicyWithCustomMaxRetries(
                                        MyDynamoDBClientParameters.maxErrorRetries));

        return clientConfiguration;
    }
}

Reply:
@poprahul Isn't DDB client config retry for the client's connection? This test is meant to retry until it successfully retrieves an item from the table.

Reply:
Yes, it is like a wait until get the flow complete and the record inserted on the table.